crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -693,6 +693,11 @@ async fn forward_software_request(
     )])
 }
 
+fn get_smartrest_template_id(payload: &str) -> Result<String, SmartRestDeserializerError> {
+    //  unwrap is safe here as the first element of the split will be the whole payload if there is no comma.
+    Ok(payload.split(',').next().unwrap().to_string())
```

Comment:
Now you don't need to return `Ok()`. Change the return type just to `String`. No reason to keep it to return `Result`.

Reply:
fixed